### ISSUE TITLE
Optimize small reads and writes 

### ISSUE THREAD

**Issue by madsbk** [rapidsai/kvikio#178]
The overhead of KvikIO becomes significant for small reads and writes.  

In cuDF we had to skip KvikIO when reading and writing small buffers:

**Comment by jakirkham** [comment]
Do we know where the overhead is coming from?

Curious if any of these might be useful:

* IOV APIs like `writelines`/`readlines` ( https://github.com/rapidsai/kvikio/issues/77 )
* Bulk `wait`ing on several `Future`s ( https://github.com/rapidsai/kvikio/issues/79 )
* Ordering of tasks with `add_done_callback` (without needing to `wait` explicitly) ( https://github.com/rapidsai/kvikio/issues/80 )

**Comment by madsbk** [comment]
No, I haven't had time to do much profiling but I think your suggestions is definitely something we should explore.
